WordPress/src/main/java/org/wordpress/android/ui/reader/subfilter/SubfilterPageFragment.kt

```diff
@@ -87,41 +80,39 @@ class SubfilterPageFragment : DaggerFragment() {
         recyclerView.isNestedScrollingEnabled = enable
     }
 
-    @Subscribe(threadMode = ThreadMode.MAIN)
-    fun onEventMainThread(event: ReaderEvents.FollowedTagsChanged) {
-        AppLog.d(T.READER, "Subfilter bottom sheet > followed tags changed")
-        viewModel.loadSubFilters()
-    }
-
-    @Subscribe(threadMode = ThreadMode.MAIN)
-    fun onEventMainThread(event: ReaderEvents.FollowedBlogsChanged) {
-        AppLog.d(T.READER, "Subfilter bottom sheet > followed blogs changed")
-        viewModel.loadSubFilters()
-    }
-
-    override fun onStart() {
-        super.onStart()
-        EventBus.getDefault().register(this)
-    }
-
-    override fun onStop() {
-        EventBus.getDefault().unregister(this)
-        super.onStop()
-    }
-
-    private fun performUpdate() {
-        performUpdate(EnumSet.of(
-                UpdateTask.TAGS,
-                UpdateTask.FOLLOWED_BLOGS
-        ))
-    }
-
-    private fun performUpdate(tasks: EnumSet<UpdateTask>) {
-        if (!NetworkUtils.isNetworkAvailable(activity)) {
+    private fun manageEmptyView(isEmpty: Boolean, category: SubfilterCategory) {
```

Comment:
This all seems like a new business logic. We might want to consider moving it into a VM and writing unit tests for it. Wdyt?

Reply:
Good point! Changed here 2930454a5828ace28737502fedfc77dbd862914f 👍!